### ISSUE TITLE
[Bug] multixy spring works only for first touch

### ISSUE THREAD

**Issue by matcham** [jean-emmanuel/open-stage-control#689]
set spring to true 
touch the multixy widget with 2 fingers.

```
(DEBUG, OSC) Out:  {
  address: '/multixy_1',
  args: [
    { type: 'f', value: 0.21276595744680862 },
    { type: 'f', value: 0.27916666666666656 },
    { type: 'f', value: 0.14680851063829795 },
    { type: 'f', value: 0.27708333333333335 }
  ]
} To: 127.0.0.1:8000
```

release 2nd touch and observe osc out message : 
nothing is sent.

release first finger : 
only when first touch is released, osc message is sent with default values : 
```
(DEBUG, OSC) Out:  {
  address: '/multixy_1',
  args: [
    { type: 'f', value: 0 },
    { type: 'f', value: 0 },
    { type: 'f', value: 0 },
    { type: 'f', value: 0 }
  ]
} To: 127.0.0.1:8000
```

I would expect spring mode to work with each touch individually.

O-S-C 1.8.8 windows

**Comment by jean-emmanuel** [comment]
Thanks for the report, this one might be hard to fix because I have no multitouch device.

Does it occur even when the 2 points are far appart from each other at release time ?

**Comment by matcham** [comment]
_Thanks for the report, this one might be hard to fix because I have no multitouch device._
hard to believe that the developer of O-S-C doesn't have a multitouch device :-)
You definitely deserve more patron supporters !

it occurs regardless the distance between the points.

**Comment by jean-emmanuel** [comment]
It should be fixed  in sources !

**Comment by jean-emmanuel** [comment]
Does it still occur in 1.8.9 ?

**Comment by matcham** [comment]
it's fixed, thanks :-)

**Comment by smithkyle** [comment]
I am seeing this again in 1.25.1 with two XY points - when I release them separately (i.e. 1 second apart), they reset to 0, but if I release them together, one of the two points does not.

**Comment by jean-emmanuel** [comment]
@smithkyle You mean two points in a single multixy widget or two xy widgets ?

**Comment by smithkyle** [comment]
Two points in a single multixy widget. When released simultaneously, only one of them springs back. If I release one shortly after the other (a half second to a second later), they both spring back. On Jul 12, 2023, at 1:29 AM, jean-emmanuel ***@***.***> wrote:﻿
@smithkyle You mean two points in a single multixy widget or two xy widgets ?

**Comment by jean-emmanuel** [comment]
I can't reproduce the issue on a galaxy tab t510 / firefox, would you mind testing again and uploading a minimal session file ?

**Comment by smithkyle** [comment]
Yes, I'm still seeing this. Here's a minimum session file, as well as a video of me recreating it on my iphone iphone 16 in safari (ios 18.3.2). I tried it in chrome on ios as well and it exhibited the same behavior.

[_TEST.json](https://github.com/user-attachments/files/19273476/_TEST.json)

https://github.com/user-attachments/assets/653e3395-f9cb-49a5-b797-ea6e552f04c5

**Comment by smithkyle** [comment]
Should also mention this is on 1.27.5

**Comment by jean-emmanuel** [comment]
Thank you, it's now fixed in sources. Note there's no good reason (that I can think of) to make a widget send directly to o-s-c's input, although in that case it helped spotted a subtle bug.

**Comment by smithkyle** [comment]
Awesome - thank you! Can you clarify what you mean by sending directly to o-s-c's input?

**Comment by jean-emmanuel** [comment]
Turns out the fix was worse than the problem, I'm reverting this change in v1.28.2 until I find a better solution.

**Comment by jean-emmanuel** [comment]
Alright I sorted it out and fixed it properly, however a similar issue still persists that's due to the UDP protocol not ensuring packets order,  it's rare but happens from to time to time (this one can't be fixed unless by using TCP or by filtering incoming feedback messages and somehow figuring if they should be ignored).
